packages/identify/src/identify.ts

```diff
@@ -148,6 +148,40 @@ export class Identify {
     return false;
   }
 
+  private _validateProperties(operation: IdentifyOperation, value: any) {
+    if (Array.isArray(value)) {
+      for (const valueElement of value) {
+        if (!(typeof valueElement === 'number' || typeof valueElement === 'string')) {
+          identifyWarn(operation, 'invalid array element type ', typeof valueElement, '. Skipping operation');
+          return false;
+        }
+      }
+    } else if (typeof value === 'object') {
+      return this._validateObject(operation, value);
+    } else if (!(typeof value === 'number' || typeof value === 'string')) {
+      identifyWarn(operation, 'invalid value type ', typeof value, '. Skipping operation');
+      return false;
+    }
+    return true;
+  }
+
+  private _validateObject(operation: IdentifyOperation, properties: { [key: string]: any }): boolean {
+    let isValidateObject = true;
+    if (Object.keys(properties).length > MAX_PROPERTY_KEYS) {
+      identifyWarn(operation, 'too many properties. Skipping operation');
+      return false;
+    }
+    for (let key in properties) {
+      if (typeof key !== 'string') {
+        identifyWarn(operation, 'invalid properties format. Skipping operation');
+        return false;
+      }
+      let value = properties[key];
+      isValidateObject = isValidateObject && this._validateProperties(operation, value);
```

Comment:
nit: you could fast-return instead of conditionally short-circuiting here - small optimization